slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -15,7 +15,7 @@ abstract class AbstractSourceCodeGenerator(model: m.Model)
       @group Basic customization overrides */
   def code = {
     "import slick.model.ForeignKeyAction\n" +
-    ( if(tables.exists(_.hlistEnabled)){
+    ( if(tables.exists(_.hlistEnabled) || tables.exists(_.columns.size >= 22)){
```

Comment:
is the second condition still needed now that we are using tuples instead of hlists for this?